java-checks/src/test/files/checks/spring/SpringComponentScanCheck.java

```diff
@@ -0,0 +1,28 @@
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Controller;
+import org.springframework.context.annotation.Import;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@ComponentScan(basePackages = "com.myapp") // Noncompliant [[sc=2;ec=15]] {{Consider replacing "@ComponentScan" by a list of beans imported with @Import to speed-up the start-up of the application.}}
+public class MyFirstApp {
+}
+
+@SpringBootApplication(scanBasePackages = "com.myapp") // Noncompliant
+class MySecondApp {
+}
+
+@SpringBootApplication // Noncompliant
```

Comment:
should add a message with `Consider replacing "@SpringBootApplication" ...` to test it gets constructed correctly